File: Convert Sorted Array to Binary Search Tree.c
struct TreeNode* convert(int* nums,int start,int end)
{
     if(start > end)
          return NULL;
     else{
         int mid = (start+end)/2;
         struct TreeNode* node = (struct TreeNode*)malloc(sizeof(struct TreeNode));
         node->val = nums[mid];
         node->left = convert(nums,start,mid-1);
         node->right = convert(nums,mid+1,end);
         return node;
     }
}

struct TreeNode* sortedArrayToBST(int* nums, int numsSize) {
    return convert(nums,0,numsSize-1);
}

File: Convert Sorted Array to Binary Search Tree.js
var sortedArrayToBST = function(nums) {
    return traveler(nums, 0, nums.length)
};

var traveler = function(temp, start, end)
{
    if(start >= end)
        return null
    else
    {
        var mid = Math.floor((start+end) / 2);
        var result = new TreeNode(temp[mid]);
        result.left = traveler(temp, start, mid)
        result.right = traveler(temp, mid+1, end)
        return result;
    }    
};
